sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/DecryptableItem.java

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.cosmos.encryption;
+
+import reactor.core.publisher.Mono;
+
+
+/// <summary>
+/// Allows for lazy decryption, which provides user a way to handle possible exceptions encountered as part of feed /
+// query processing.
+/// Also provides decryption operation details.
+/// </summary>
+/// <example>
+/// <code language="c#">
+/// <![CDATA[
+/// public class ToDoActivity{
+///     public string id {get; set;}
+///     public string status {get; set;}
+///     public int cost {get; set;}
+/// }
+///
+/// QueryDefinition queryDefinition = new QueryDefinition("select * from ToDos");
+/// using (FeedIterator<DecrytableItem> feedIterator = this.Container.GetItemQueryIterator<DecrytableItem>(
+///     queryDefinition,
+///     requestOptions: new QueryRequestOptions() { PartitionKey = new PartitionKey("Error")}))
+/// {
+///     while (feedIterator.HasMoreResults)
+///     {
+///         FeedResponse<DecryptableItem> decryptableItems = await feedIterator.ReadNextAsync();
+///         foreach(DecryptableItem item in decryptableItems){
+///         {
+///             try
+///             {
+///                 (ToDoActivity toDo, DecryptionContext _) = await item.GetItemAsync<ToDoActivity>();
+///             }
+///             catch (EncryptionException encryptionException)
+///             {
+///                 string dataEncryptionKeyId = encryptionException.DataEncryptionKeyId;
+///                 string rawPayload = encryptionException.EncryptedContent;
+///             }
+///         }
+///     }
+/// }
+/// ]]>
+/// </code>
+/// </example>
+
+// TODO: moderakh add example
```

Comment:
TODOs - are they tracked in a separate PR? or can this todo be removed due to the example provided above?

Reply:
the public APIs haven't finalized yet. For that reason I would like to leave wiring the examples for later (to avoid re-do work), once the final APIs are finalized.